File: Arrays/Medium/11/11.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        max_area = 0

        while left < right:
            h = min(height[left], height[right])
            w = right - left
            area = h * w
            max_area = max(max_area, area)

            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return max_area
    
  # Two-pointer method , going from left and right
  # Iterative approach to constantly check max area
  # Finally return max area when loop finishes

File: Arrays/Medium/11/11.ts
function maxArea(height: number[]): number {
  let left: number = 0;
  let right: number = height.length - 1;
  let maxArea: number = 0;

  while (left < right) {
    const h: number = Math.min(height[left], height[right]);
    const w: number = right - left;
    const area: number = h * w;
    maxArea = Math.max(maxArea, area);

    if (height[left] < height[right]) {
      left++;
    } else {
      right--;
    }
  }

  return maxArea;
}
